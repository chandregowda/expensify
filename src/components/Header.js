import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => {
	return (
		<header>
			<h1>Expensify</h1>
			<nav className="nav-links">
				<NavLink exact activeClassName="is-active" to="/">
					Home
				</NavLink>
				<NavLink exact activeClassName="is-active" to="/add">
					Add
				</NavLink>
				<NavLink exact activeClassName="is-active" to="/help">
					Help
				</NavLink>
			</nav>
		</header>
	);
};

export default Header;
