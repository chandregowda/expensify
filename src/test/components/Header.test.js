import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../components/Header';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

test('Should render Header Component', () => {
	const warpper = shallow(<Header />);
	expect(warpper.find('h1').length).toBe(1);
	expect(warpper.find('h1').text()).toBe('Expensify');
	expect(warpper).toMatchSnapshot();
	// const renderer = new ReactShallowRenderer();
	// renderer.render(<Header />);
	// expect(renderer.getRenderOutput()).toMatchSnapshot();
});
