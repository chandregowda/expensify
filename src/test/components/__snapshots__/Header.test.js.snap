// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Header Component 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <nav
    className="nav-links"
  >
    <NavLink
      activeClassName="is-active"
      ariaCurrent="true"
      exact={true}
      to="/"
    >
      Home
    </NavLink>
    <NavLink
      activeClassName="is-active"
      ariaCurrent="true"
      exact={true}
      to="/add"
    >
      Add
    </NavLink>
    <NavLink
      activeClassName="is-active"
      ariaCurrent="true"
      exact={true}
      to="/help"
    >
      Help
    </NavLink>
  </nav>
</header>
`;
